src/cmds/pbsnodes.c

```diff
@@ -362,7 +362,7 @@ encode_to_json(struct batch_status *bstat) {
 					return 1;
 			} else {
 				node->value_type = JSON_STRING;
-				node->value.string = strdup(str);
+				node->value.string = strdup_escape(str);
```

Comment:
Hi @vstumpf ,
Many lines of code in encode_to_json() are just a duplication of add_json_node() functionality. Instead of invoking strdup_escape(), Can we try to call add_json_node which will handle most of the functionality including identifying the json type. Your approach is simple, but for future maintenance we may need to make changes in multiple places which otherwise will just go in the library function.

Reply:
I can only see one part of `encode_to_json` that can be replaced by a better `add_to_json` call. The others are necessary due to special cases like arrays and only non-zeros. Since there's no easy way to remove a json node, I'll keep most of it as it was.

Reply:
I would love to see `node` to be treated as a `private` member only getting manipulated through associated functions. I think we can still do that for the last `else` block with special treatment only if the attribute name is jobs.

Reply:
I can change it, but that'd increase the scope of this pr, which was originally for the nonconforming json in the comment value. Should there be another ticket for that?